File: #0209.minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {

        int size = nums.size();
        int left = 0;
        int right = 0;
        int sum = 0;
        int result = size + 1;
        while (right < size) {
            while (sum < target && right < size) {
                sum += nums[right++];
            }
            while (sum >= target) {
                result = std::min(result, right - left);
                sum -= nums[left++];
            }
        }
        return result == size + 1 ? 0 : result;
    }
};

File: #0209.minimum-size-subarray-sum.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(target, nums) {
    let sum = 0;
    let start = 0;
    let minLength = Infinity;
    for (let end = 0; end < nums.length; end++) {
        sum += nums[end];
        while (sum >= target) {
            minLength = Math.min(minLength, end - start + 1);
            sum -= nums[start];
            start++;
        }
    }
    return (minLength === Infinity) ? 0 : minLength;
};
